### DISCUSSION TITLE
Tests failing after compilation and generation of executable in an HPC system

### DISCUSSION THREAD

**Discussion by arjunvarmar** [idaholab/moose#27109] [Q&A General] [answered]
[run_tests_log.txt](https://github.com/idaholab/moose/files/14621167/run_tests_log.txt)
I am trying to run MOOSE on an HPC system. I have finished the installation process as per the documentation and generated the `run_tests` executable in the `test` folder in `moose`. However, when I run the tests, only 2 are successful and the rest fail. 

I am attaching the output log of the run as a txt file herewith. 

One of the errors that I see here (and also when I submit specific jobs using an example input file) is that the parameters mentioned in the input files are supplied twice. 

For instance,

`/misc/check_error/bad_kernel_var_test.i:41.5: parameter 'BCs/DirichletBC_type/boundary' supplied multiple times`

I was wondering if anyone has come across this issue before or knows how to resolve it. Any help would be much appreciated.

**Comment by brandonlangley** [comment, marked as answer]
@arjunvarmar -

I see from looking at your test output log that the problem is your WASP package has an incorrect configuration.

Specifically, the `DISABLE_HIT_TYPE_PROMOTION` CMake variable is turned off for your WASP build when it should be turned on.

Are you building MOOSE by using the pre-built Conda packages for WASP (and libMesh and PETSc)?

Or are you building your own WASP by using the `scripts/update_and_rebuild_wasp.sh` script?

**Reply by brandonlangley** [reply]
@arjunvarmar -

I missed where you said this is a HPC system, so I assume you are building your own WASP and not using the Conda build.

Are you calling the `scripts/update_and_rebuild_wasp.sh` script to build WASP?

This script should use a configuration that turns on `DISABLE_HIT_TYPE_PROMOTION`, but that is not what your failures show.

**Reply by arjunvarmar** [reply]
Thanks a lot for that input. I was trying to use a pre-built module of WASP. When I used the `scripts/update_and_rebuild_wasp.sh` build instead, the tests ran fine! Thanks again.
